packages/connect-popup/e2e/tests/passphrase.test.ts

```diff
@@ -0,0 +1,122 @@
+    ]);
+
+    await popup.waitForSelector("button[data-test='@analytics/continue-button']", {
+        state: 'visible',
+        timeout: 40000,
+    });
+    await popup.click("button[data-test='@analytics/continue-button']");
+
+    await popup.waitForSelector("button[data-test='@permissions/confirm-button']", {
+        state: 'visible',
+    });
+
+    await popup.click("button[data-test='@permissions/confirm-button']");
+    await popup.click("button[data-test='@export-address/confirm-button']");
+
+    (await popup.waitForSelector("input[data-test='@passphrase/input']")).type('abc');
+
+    await popup.click("button[data-test='@passphrase/hidden/submit-button']");
+    // Accept to see Passphrase.
+    await TrezorUserEnvLink.api.pressYes();
+
+    // Confirm Passphrase is correct.
+    await TrezorUserEnvLink.api.pressYes();
+
+    // Confirm right address is displayed.
+    await TrezorUserEnvLink.api.pressYes();
+
+    await popup.waitForSelector("button[data-test='@connect-ui/success-close-button']");
+
+    await popup.click("button[data-test='@connect-ui/success-close-button']");
+});
+
+test('introduce passphrase in popup and device rejects it', async ({ page }) => {
+    await TrezorUserEnvLink.api.stopBridge();
+    await TrezorUserEnvLink.api.stopEmu();
+    await TrezorUserEnvLink.api.startEmu({
+        wipe: true,
+    });
+    await TrezorUserEnvLink.api.setupEmu({
+        mnemonic: 'alcohol woman abuse must during monitor noble actual mixed trade anger aisle',
+        pin: '',
+        passphrase_protection: true,
+        label: 'My Trevor',
+        needs_backup: false,
+    });
+    await TrezorUserEnvLink.api.startBridge(bridgeVersion);
+
+    await page.goto(`${url}#/method/getAddress`);
+    await page.waitForSelector("button[data-test='@submit-button']", { state: 'visible' });
+
+    [popup] = await Promise.all([
+        page.waitForEvent('popup'),
+        page.click("button[data-test='@submit-button']"),
+    ]);
+
+    await popup.waitForSelector("button[data-test='@analytics/continue-button']", {
+        state: 'visible',
+        timeout: 40000,
+    });
+    await popup.click("button[data-test='@analytics/continue-button']");
+
+    await popup.waitForSelector("button[data-test='@permissions/confirm-button']", {
+        state: 'visible',
+    });
+
+    await popup.click("button[data-test='@permissions/confirm-button']");
+    await popup.click("button[data-test='@export-address/confirm-button']");
+
+    (await popup.waitForSelector("input[data-test='@passphrase/input']")).type('abc');
+
+    await popup.click("button[data-test='@passphrase/hidden/submit-button']");
+    // Accept to see Passphrase.
+    await TrezorUserEnvLink.api.pressYes();
+
+    // Confirm Passphrase is correct.
+    await TrezorUserEnvLink.api.pressNo();
+
+    await popup.waitForSelector("div[data-test='@connect-ui/error']", {
+        state: 'visible',
+        timeout: 40000,
```

Comment:
maybe timeout shouldn't be that long so  that we validate that the error screen really opens within reasonable time?

Reply:
Yes, that makes sense. Changed.